lib/ClangImporter/ImportType.cpp

```diff
@@ -2438,10 +2439,14 @@ static ModuleDecl *tryLoadModule(ASTContext &C,
 
   // If we're synthesizing forward declarations, we don't want to pull in
   // the module too eagerly.
-  if (importForwardDeclarations)
+  if (importForwardDeclarations) {
     module = C.getLoadedModule(moduleName);
-  else
+  } else {
     module = C.getModuleByIdentifier(moduleName);
+    if (module) {
+      (void) namelookup::getAllImports(module);
+    }
```

Comment:
Not knowing the code, I can't see how it follows from the comment on 2440-2441.

Reply:
This is the unstated "otherwise" case here. We're *not* synthesizing forward declarations here, we're pulling in definitions. And because of the legacy behavior of the module loader, you can wind up with situations where the import behavior of Clang modules depends on overlays being loaded. At the very least you need Foundation's overlay loaded to get many automatic bridging behaviors.